docs/design/AgentRun-CLI-Design.md

```diff
@@ -0,0 +1,658 @@
+- Packaging the workspace and metadata into a cloud-compatible format
+- Setting up required roles, permissions, and credentials for the cloud build service
+
+**Step 2:**  After packaging the agent workspace, the developer initiates the build process using: 
+
+```agentrun build -f ./```
+
+This command triggers the image build based on the workspace contents and metadata configuration. The CLI supports an optional `-p` flag to specify a **custom proxy**, which is particularly useful for environments with restricted network access or internal mirrors.
+- For **Python agents**, the proxy is applied to `pip` commands during dependency installation   
+- For **Java agents**, the proxy is applied to `mvn` (Maven) commands during build resolution
+
+**Step 2.1** Build Service Validation
+The `agentrun build` command is handled by the **Build** service, which performs a series of validation checks to ensure the build process can succeed in both **local** and **cloud** modes.
+
+Key validations include:
+- **Workspace integrity**: Verifies that the agent workspace is correctly structured and includes all required files (e.g., source code, metadata, dependencies).
+- **Metadata consistency**: Confirms that runtime metadata matches the expected format and values needed for image generation.
+- **Build mode compatibility**:
+    - For **local builds**, the system checks whether Docker or Podman is installed and accessible on the developer’s machine.    
+    - For **cloud builds**, the service validates credentials, roles, and connectivity to the configured cloud build provider.
+
+**Step 2.1.1** Local Build
+In local build mode, the CLI invokes the local container runtime (Docker or Podman) to build the agent image. The image is tagged using the agent name defined in the metadata configuration file, ensuring traceability and consistency.
+
+**Step 2.1.2** Cloud Build
+TBD
+
+**Step 3** Once the agent image is successfully built, the developer can publish it to AgentCube using:
+
+```agentrun publish -f ./```
+
+This command initiates the publishing process, registering the agent with AgentCube and making it available for invocation, collaboration, and public or team access.
+
+The behavior of the publish process depends on the **build** mode. It requires options:
+- **Image Repository URL and Credentials**
+    - **Local build mode**: The developer must explicitly provide the image repository URL, along with username and password, via CLI options.
+    - **Cloud build mode**: The image is already built and stored in the cloud image repository. The image location is automatically retrieved and merged into the metadata configuration file during the build process.
+- **Agent Version**
+    - A semantic version string (e.g., `v1.0.0`) used to register the agent.
+    - If publishing a new version, the version must be unique.
+    - If updating an existing version, the same version number can be reused.
+- **Additional Metadata**
```

Comment:
If this can be specified with `agentrun publish`, please give an example on how to use the flags